### PR TITLE
feat: provide data to component

### PR DESCRIPTION
**Pull request by rhutchison**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #74 

## What is the new behavior?

When using tippy directive with a custom component, we want to be able to pass the data input in to the custom component through the TIPPY_REF.

```
constructor(@Inject(TIPPY_REF) tippy: TippyInstance) {
  console.log(tippy.data);
}
```

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/app/playground/playground.component.html [hunk 1]
```diff
@@ -112,8 +112,21 @@ <h6>ng-container</h6>
...
+        <input type="text" #input (keyup)="(true)" value="ngneat" />
+      </label>
+    </div>
+
     <div class="btn-container">
-      <button class="btn btn-outline-danger" [tippy]="comp" variation="popper" (visible)="handleStatus($event)">
+      <button
+        class="btn btn-outline-danger"
+        [tippy]="comp"
+        [data]="{ name: input.value }"
```
Comment (NetanelBasal) [inline src/app/playground/playground.component.html:126]: It's important to mention that it's not "reactive".

**Review by NetanelBasal — APPROVED** [review]

**Comment by rhutchison** [comment]
@NetanelBasal are you ok with adding `data?: any` to CompViewOptions? https://github.com/ngneat/overview/blob/main/projects/ngneat/overview/src/lib/views/view.ts#L20

**Comment by NetanelBasal** [comment]
For what purpose?

**Comment by rhutchison** [comment]
> For what purpose?

https://github.com/ngneat/helipopper/pull/79/files#diff-f9b629987d3e3c4f5011af27a3699b019b08ae0b24c8d13b9393180b6520167dR94

**Comment by NetanelBasal** [comment]
No, I don't think it should be there. You're the one responsible for passing an injector.

**Comment by rhutchison** [comment]
> No, I don't think it should be there. You're the one responsible for passing an injector.

How do you recommend passing the data to the service directly?  https://github.com/ngneat/helipopper/blob/master/projects/ngneat/helipopper/src/lib/tippy.service.ts#L16

I could have addressed the directive only, but the PR feel incomplete without the service working also.

**Comment by NetanelBasal** [comment]
Add it to the instance: https://github.com/ngneat/helipopper/blob/master/projects/ngneat/helipopper/src/lib/tippy.service.ts#L30

**Comment by rhutchison** [comment]
> Add it to the instance: https://github.com/ngneat/helipopper/blob/master/projects/ngneat/helipopper/src/lib/tippy.service.ts#L30

I did: https://github.com/ngneat/helipopper/pull/79/files#diff-63acb732814618af8b7a6b950efaa94c3443fff425de964d3c009de47d9186c0R30-R38

But I can't add it to the instance unless I pass it through the options, which is the issue here.  I am open to another approach, if there is one.

**Comment by NetanelBasal** [comment]
You need to update this type: https://github.com/ngneat/helipopper/blob/3e8ac2c91666894b3e0f9c2f023d2301c34132af/projects/ngneat/helipopper/src/lib/tippy.types.ts#L5

**Comment by davayd** [comment]
🙌⚡

**Pull request merged**